agent/provider/src/provider_agent.rs

```diff
@@ -66,6 +67,56 @@ impl GlobalsManager {
     }
 }
 
+struct WhitelistManager {
+    state: DomainWhitelistState,
+    monitor: Option<FileMonitor>,
+}
```

Comment:
Please add a short comment documenting what this structure does.

Reply:
Adding type named "manager" always feels like admitting to a design/naming effort defeat (in my defense there are other "managers" nearby it). 
Will add a description.
Done.